packages/insomnia-app/app/common/analytics.ts

```diff
@@ -129,6 +127,17 @@ export enum SegmentEvent {
   unitTestDelete = 'Unit Test Deleted',
   unitTestRun = 'Ran Individual Unit Test',
   unitTestRunAll = 'Ran All Unit Tests',
+  vcsSyncStart = 'VCS Sync Started',
+  vcsSyncComplete = 'VCS Sync Completed',
+  vcsAction = 'VCS Action Executed',
+}
+
+export function vcsSegmentEventProperties(type: String, action: String, error?: string): Record<string, any> {
```

Comment:
We can remove the return type and TS will automatically interpret it from the implementation 🎉 we can also be more specific with the allowed strings for each parameter so that we have type-safety in case of typos etc.

Two further suggested changed below to keep TS happy

```suggestion
type PushPull = 'push' | 'pull';

export function vcsSegmentEventProperties(
  type: 'git',
  action: PushPull | `force_${PushPull}` | 'create_branch' | 'merge_branch' | 'delete_branch' | 'checkout_branch' | 'commit' | 'stage_all' | 'stage' | 'restore' | 'update' | 'setup' | 'clone',
  error?: string
) {
```